source/bases/Win32Service.c

```diff
@@ -494,10 +499,11 @@ static int Service_Uninstall(
         return LogWin32Error(GetLastError(), "cannot open service manager");
 
     // create service
-    serviceHandle = OpenServiceW(managerHandle, PyUnicode_AS_UNICODE(fullName),
-            SERVICE_ALL_ACCESS);
+    wfullName = PyUnicode_AsWideCharString(fullName, NULL);
+    serviceHandle = OpenServiceW(managerHandle, wfullName, SERVICE_ALL_ACCESS);
     if (!serviceHandle)
         return LogWin32Error(GetLastError(), "cannot open service");
+    PyMem_Free(wfullName);
```

Comment:
This should happen before the check on the service handle; otherwise, a potential memory leak occurs!

Reply:
Only when compiling via sources will this happen, not in installations in general. For example, even if I had instructions, warning, or documentation I would prefer it to be automated and I think whoever opened the issue too. When I went to test issue #536, I had a good job to find out how to use cx_Logging, if it were automatic, it would be so simple.

Reply:
This leak and others resolved.